### PR TITLE
Bug 52744 proxy timeout

### PR DESCRIPTION
**Pull request by Thunderguts13**

### CODE REVIEW

File: app/models/maintenance_tasks/removed_ldap_users.rb [hunk 1]
```diff
@@ -32,12 +31,12 @@ def enabled?
   private
 
   def collect_removed_ldap_users
-    User::Human.ldap.collect do |user|
-      user unless ldap_connection.exists?(user.username)
-    end.compact
+    User::Human.ldap.reject do |user|
+      ldap_connection.all_uids.include?([user.username])
```
Comment (TheWalkingLeek) [inline app/models/maintenance_tasks/removed_ldap_users.rb]: Is this a two dimensional array? Is that why you include on an array? Would it make more sense to flatten it somewhere if there's only the username in the array?

File: lib/ldap_connection.rb [hunk 2]
```diff
@@ -178,4 +173,10 @@ def search_for_login(username)
     connection.search(base: basename, filter: filter) { |entry| ldap_entry = entry }
     ldap_entry
   end
+
+  def fetch_all_uids
+    filter = Net::LDAP::Filter.pres('uid')
+    users = connection.search(base: basename, filter: filter, attributes: 'uid')
+    users.map(&:uid)
```
Comment (TheWalkingLeek) [inline lib/ldap_connection.rb]: Nice!

File: test/unit/models/maintenance_tasks/removed_ldap_users_test.rb [hunk 3]
```diff
@@ -54,19 +53,13 @@ class RemovedLdapUsersTest < ActiveSupport::TestCase
...
-      expects(:exists?).
-      with('bob').
-      returns(false)
-
-    LdapConnection.any_instance.
-      expects(:exists?).
-      with('alice').
-      returns(true)
+      expects(:all_uids).
+      returns([['alice']]).
```
Comment (TheWalkingLeek) [inline test/unit/models/maintenance_tasks/removed_ldap_users_test.rb]: Same here as above

File: app/models/maintenance_tasks/removed_ldap_users.rb (hunk 1 above)
Comment (Thunderguts13) [reply app/models/maintenance_tasks/removed_ldap_users.rb]: Yes you're right I didn't think of that :smile:

**Pull request merged**